### ISSUE TITLE
unused_unit will remove items led by attributes

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#9949]
### Summary

.

### Lint Name

unused_unit

### Reproducer

I tried this code:

```rust
fn main() {
    #[doc = "trolol"]
    ()
}
```

clippy will suggest to remove the `()` which will cause a compiler error because we now have an attribute that is not attached to any item :/ 

````
This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error: expected statement after outer attribute
 --> src/main.rs:2:5
  |
2 |     #[doc = "trolol"]
  |     ^^^^^^^^^^^^^^^^^

error: aborting due to previous error

Original diagnostics will follow.

warning: unused doc comment
 --> src/main.rs:2:5
  |
2 |     #[doc = "trolol"]
  |     ^^^^^^^^^^^^^^^^^
3 |     ()
  |     -- rustdoc does not generate documentation for expressions
  |
  = help: use `//` for a plain comment
  = note: `#[warn(unused_doc_comments)]` on by default

warning: unneeded unit expression
 --> src/main.rs:3:5
  |
3 |     ()
  |     ^^ help: remove the final `()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unused_unit
  = note: `#[warn(clippy::unused_unit)]` on by default
````

### Version

```text
rustc 1.67.0-nightly (b3bc6bf31 2022-11-24)
binary: rustc
commit-hash: b3bc6bf31265ac10946a0832092dbcedf9b26805
commit-date: 2022-11-24
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
